src/components/implementation/capmgr/naive/mem_mgr.c

```diff
@@ -65,3 +65,35 @@ memmgr_shared_page_map_cserialized(vaddr_t *pgaddr, int *unused, int idx)
 done:
 	return num_pages;
 }
+
+/* These functions return the address at which the gs register is set to */
+void *
+memmgr_tls_alloc(unsigned int dst_tid)
```

Comment:
Why is it called `alloc()` if it's not doing any `allocation`? 
Why not use the type `thdid_t` here? 

Shouldn't we check if the client (calling) component has been initialized with `capmgr` ??

Reply:
"Shouldn't we check if the client (calling) component has been initialized with capmgr ??" - We should, do you have an example of where we do that currently?

The name choice of alloc is bad.

Reply:
`if the client (calling) component has been initialized with capmgr` is checked in every existing interface implementation API in capmgr. 

For ex (from `memmgr_heap_page_allocn()`):
```
        spdid_t cur = cos_inv_token();
        struct cos_compinfo  *cap_ci  = cos_compinfo_get(cos_defcompinfo_curr_get());
        struct cap_comp_info *cur_rci = cap_info_comp_find(cur);
        struct cos_compinfo  *cur_ci  = cap_info_ci(cur_rci);
        vaddr_t src_pg, dst_pg;
        unsigned int off = 0;

        if (!cur_rci || !cap_info_init_check(cur_rci)) return 0;
        if (!cur_ci) return 0;
```